packages/components/src/InputFile/InputFile.tsx

```diff
@@ -258,7 +293,12 @@ export function InputFile({
       file,
       handleUploadProgress,
     });
-    axios.request(axiosConfig).then(handleUploadComplete);
+    axios
+      .request(axiosConfig)
+      .then(handleUploadComplete)
+      .catch(() => {
+        onUploadError && onUploadError(new Error("Failed to upload file"));
```

Comment:
thinking out loud here, might not require any changes

we have the `validator` which will display messages when a file is rejected, and that's done inside this component

then we have the `onUploadError` that will trigger if the upload fails either from getting the upload params or if the actual request fails. however neither of these have any built in visual communication that something went wrong, it would be left up to each consumer how to communicate that - if at all

I feel conflicted about that. I'm not opposed to the consumer being informed and being able to do whatever makes sense for them, but I wonder about having a consistent baseline experience when the upload fails that at least lets the user know something went wrong

Reply:
or phrased another way, should visual error states be a concern of this component in the same way the validations are?

Reply:
I think that's more of a design decision for @chris-at-jobber. One thing I'm thinking about is that when the upload params and request fail it's more of a system error. In the case of the user uploading the wrong file type, that is something more in their control. I can add props and methods to keep track of that those errors but I'm a bit worried that it might blow up in scope.

Reply:
agreed on both a design decision and scope adding

I'd say hold off on adding anything until we have a concrete decision of "yeah we need that". again this was some thinking out loud, and not a demand/request!